### PR TITLE
(#21568): adding support for PPing plugins

### PR DESCRIPTION
**Pull request by victoralfaro-dotcms**
Adding manifest file for plugin PP to avoid XML bundle writing, centralizing ObjectMapper and XStream instances to one and adding plugin case to Dependency manager.

Sibling PRs:
https://github.com/dotCMS/enterprise/pull/854

### CODE REVIEW

**Review by freddyucv — COMMENTED** [review]
In the PP performance changes we move the XStream local variable into a class field to reuse it because it was taking a lot time create a XStream instance each time we need it

https://github.com/dotCMS/core/commit/47841c7d0de6d8b0f6eec558c6b17a36627a4fed

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/publishing/XMLSerializerUtil.java [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
-import java.io.OutputStream;
-import java.io.OutputStreamWriter;
 
 /**
  * Util for serialize object to XML file
  */
 public class XMLSerializerUtil {
 
     private static final XMLSerializerUtil instance= new XMLSerializerUtil();
-    private XStream xmlSerializer = null;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/publishing/XMLSerializerUtil.java:17]: It seems you still using XStream

Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/publishing/XMLSerializerUtil.java:17]: @jdotcms I need to. For publishing purposes it serializes to a XML representation of several objects. Mainly `PublishAuditHistory` and `PushPublisherConfig`. I switched to JSON just for the manifest.

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/publisher/business/PublishAuditHistory.java [hunk 2]
```diff
@@ -115,8 +115,8 @@ public void addOrUpdateEndpoint(String groupId, String endpointId, EndpointDetai
 	
 	static XStream xstream=new XStream(new DomDriver());
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/business/PublishAuditHistory.java:116]: why using Xstream?

File: dotCMS/src/main/java/com/dotcms/publishing/XMLSerializerUtil.java [hunk 3]
```diff
@@ -28,7 +28,11 @@ public static XMLSerializerUtil getInstance() {
     }
 
     private XMLSerializerUtil(){
-        xmlSerializer = new XStream(new DomDriver("UTF-8"));
+        xmlSerializer = new XStream(new DomDriver(StandardCharsets.UTF_8.name()));
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/publishing/XMLSerializerUtil.java:31]: Jackson can generate the XML, can not?

**Pull request closed without merging**